### DISCUSSION TITLE
Failed on next/route installation.‏

### DISCUSSION THREAD

**Discussion by EliyaUYosef** [vercel/vercel#10810] [Help]
### Question

Hey, I'm Eliya Yosef,

I'm encountering some issues with the installation of `next/route`. I've provided the error message below after attempting to install it:

❯ npm install next/route --loglevel verbose

npm verb cli /usr/local/bin/node /usr/local/bin/npm
npm info using npm@9.2.0
npm info using node@v19.3.0
npm verb title npm install next/route
npm verb argv "install" "next/route" "--loglevel" "verbose"
npm verb logfile logs-max:10 dir:/Users/eliyayosef/.npm/_logs/2023-11-08T18_19_59_306Z-
npm verb logfile /Users/eliyayosef/.npm/_logs/2023-11-08T18_19_59_306Z-debug-0.log
npm verb stack Error: An unknown git error occurred
npm verb stack at makeError (/usr/local/lib/node_modules/npm/node_modules/@npmcli/git/lib/make-error.js:28:13)
npm verb stack at /usr/local/lib/node_modules/npm/node_modules/@npmcli/git/lib/spawn.js:37:26
npm verb cwd /Users/eliyayosef/Documents/my_tasks/client
npm verb Darwin 21.6.0
npm verb node v19.3.0
npm verb npm v9.2.0
npm ERR! code 128
npm ERR! An unknown git error occurred
npm ERR! command git --no-replace-objects ls-remote ssh://[git@github.com/next/route.git](http://git@github.com/next/route.git)
npm ERR! ERROR: Repository not found.
npm ERR! fatal: Could not read from remote repository.
npm ERR!
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm verb exit 128
npm verb unfinished npm timer reify 1699467599383
npm verb unfinished npm timer reify:loadTrees 1699467599388
npm verb unfinished npm timer idealTree:userRequests 1699467599520
npm verb code 128

npm ERR! A complete log of this run can be found in:
npm ERR! /Users/eliyayosef/.npm/_logs/2023-11-08T18_19_59_306Z-debug-0.log

I also checked my SSH authentication with GitHub and received the following confirmation:

❯ ssh -T [git@github.com](mailto:git@github.com)

Hi EliyaUYosef! You've successfully authenticated, but GitHub does not provide shell access.

I'd appreciate your assistance with this. 🙏🏻

All I need is to use the `router.push('/login')` line, but I'm encountering difficulties with the installation of `next/route`. If you have any insights or solutions, please let me know. Thank you!

more info : 
  ❯ node -v
    v19.3.0
  ❯ npm -v
    9.2.0 

for more info, i'm here 😉

**Comment by mikedidomizio** [comment]
Hey @EliyaUYosef

That package doesn't exist (or is private).  NPM appears to default to looking for the repository if the package isn't found in the registry.  You can see the same error if you do something like `npm install next/gibberish`.

If I understand correctly, the router you're looking for is part of your Next installation already.  If you're using Next with the app router here is some [documentation](https://nextjs.org/docs/app/api-reference/functions/use-router), if you're using Next with the pages router, here is some [documentation](https://nextjs.org/docs/pages/api-reference/functions/use-router).

**Reply by EliyaUYosef** [reply]
Thank you for your help! I appreciate you taking the time to explain this to me. I missed that in the documentation. You were a big help in solving my problem. Using import { useRouter } from 'next/navigation' instead of import { useRouter } from 'next/route' fixed the issue. 🙌
